os/fs/driver/mtd/mtd_jedec.c

```diff
@@ -535,6 +536,70 @@ static inline void jedec_bytewrite(struct jedec_dev_s *priv, FAR const uint8_t *
 }
 #endif
 
+/************************************************************************************
+ * Name: jedec_is_erased
+ ************************************************************************************/
+static int jedec_is_erased(FAR struct jedec_dev_s *priv, off_t block)
+{
+	size_t offset;
+	size_t count;
+	uint32_t *buffer;
+	uint32_t erasesize;
+	jedec_geometry_t geo;
+	int ret = 1;
+	
+	DEBUGASSERT(priv);
+	geo = priv->geo;
+
+	if (geo.subsectorshift > 0) {
+		erasesize = (1 << geo.subsectorshift);
+	} else {
+		erasesize = (1 << geo.sectorshift);
+	}
+
+	buffer = kmm_malloc(erasesize * sizeof(uint8_t));
```

Comment:
can we avoid multiple times allocation and free, as called in while (blocksleft > 0)

Reply:
yeah good idea, but most of case this being called just one time. so I will keep current state, but I will improve whole of structure in future..